### PR TITLE
Draw mesh color in SimpleFaceGraphViewerQt

### PR DESCRIPTION
**Pull request by maximecharriere**
**Concern:** SimpleFaceGraphViewerQt class & Surface Mesh package  

## Explanation
Currently, SimpleFaceGraphViewerQt shows a mesh with random face colors (when `mono-color` is disabled), even if this mesh countain color's property map in his properties.   
It seems more logical to me that the colors present in the color property map are shown, and only if this property is non-existent, a random colour is assigned. 

## Summary of Changes
- Use SurfaceMesh's color property map to color the drawed element. If none, use a random color ([R47](https://github.com/CGAL/cgal/compare/master...maximecharriere:dev/drawGraphColor?expand=1#diff-a70b0eabf501d954e265c41ac95cb64be1f6f1e67a2ca79cbb9881d9f2f8139eR47)),
- Use default `f:color`/`v:color`/`e:color` property map names to found colors ([R25](https://github.com/CGAL/cgal/compare/master...maximecharriere:dev/drawGraphColor?expand=1#diff-a70b0eabf501d954e265c41ac95cb64be1f6f1e67a2ca79cbb9881d9f2f8139eR25-R36)),
- Colors not only faces, but also vertex and edges ([R177](https://github.com/CGAL/cgal/compare/master...maximecharriere:dev/drawGraphColor?expand=1#diff-a70b0eabf501d954e265c41ac95cb64be1f6f1e67a2ca79cbb9881d9f2f8139eR177)),
- Remove the mono-color generation because it seems to me that it has no effect. Mono-color management is handled by `Basic_viewer_qt` ([L33](https://github.com/CGAL/cgal/compare/master...maximecharriere:dev/drawGraphColor?expand=1#diff-a70b0eabf501d954e265c41ac95cb64be1f6f1e67a2ca79cbb9881d9f2f8139eL33))

### CODE REVIEW

**Comment by gdamiand** [comment]
Thanks @maximecharriere for the patch. I agree to integrate it.

**Review by gdamiand — APPROVED** [review]

File: BGL/include/CGAL/draw_face_graph.h [hunk 1]
```diff
@@ -22,17 +22,29 @@
...
+  CGAL::IO::Color operator()(const Graph& mesh,
+                         EI elementIndex) const
   {
-    if (fh==boost::graph_traits<Graph>::null_face()) // use to get the mono color
-      return CGAL::IO::Color(100, 125, 200); // R G B between 0-255
-
-    return get_random_color(CGAL::get_default_random());
+    typename Graph::template Property_map<EI, CGAL::IO::Color> colorPm;
+    bool found;
+    std::tie(colorPm, found) = mesh.property_map<EI, CGAL::IO::Color>(getColorPropertyName<EI>()); //Get the color property map
```
Comment (sloriot) [inline BGL/include/CGAL/draw_face_graph.h:46]:
```suggestion
    std::tie(colorPm, found) = mesh.template property_map<EI, CGAL::IO::Color>(getColorPropertyName<EI>()); //Get the color property map
```

**Comment by sloriot** [comment]
The code added is specific to `Surface_mesh`. [Dynamic properties](https://doc.cgal.org/latest/BGL/index.html#title25) should be used

**Comment by sloriot** [comment]
```
/home/cgal_tester/build/src/cmake/platforms/Debian-testing-clang-main/test/Surface_mesh_Examples/draw_surface_mesh.cpp:26:98: warning: expression result unused [-Wunused-value]
  /*auto fcm =*/ sm.add_property_map<Mesh::Face_index>("f:color", CGAL::IO::white() /*default*/).first;
  ```

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-169

**Pull request merged**